### PR TITLE
[openshift] Add a custom SCC to be bound to default service account

### PR DESCRIPTION
**Pull request by nikhil-thomas**
Add a `pipelines-scc` SecurityContextConstraint to be used with default
pipleine service account `pipeline`.

This solves the issue of `git-clone` task (or any task up ahead in pipeline) blocking
file permissions for subsequent tasks.

Add a mechanism to cleanup pipeline-anyuid rolebinding

Add a cleanup mechanism in openshift exetension for TektonConfig
reconciler to delete `pipeline-anyuid` rolebinding from relevant namespaces.

Add mechanism to remove `pipeline-anyuid` clusterrole

OpenShift specific change, therefore no release note.

Signed-off-by: Nikhil Thomas <nikthoma@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: cmd/openshift/operator/kodata/tekton-pipeline/00-prereconcile/openshift-pipelines-scc.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+allowHostDirVolumePlugin: false
+allowHostIPC: false
+allowHostNetwork: false
+allowHostPID: false
+allowHostPorts: false
+allowPrivilegeEscalation: true
+allowPrivilegedContainer: false
+allowedCapabilities: null
+apiVersion: security.openshift.io/v1
```
Comment (vdemeester) [inline cmd/openshift/operator/kodata/tekton-pipeline/00-prereconcile/openshift-pipelines-scc.yaml:9]: Can we reorder this file just a but ? 

```yaml
apiVersion: …
kind: …
metadata:
  …
…
```

**Review by vdemeester — COMMENTED** [review]
@nikhil-thomas this doesn't remove the old `pipeline-anyuid` though right ?

File: cmd/openshift/operator/kodata/tekton-pipeline/00-prereconcile/openshift-pipelines-scc.yaml (hunk 1 above)
Comment (chmouel) [reply cmd/openshift/operator/kodata/tekton-pipeline/00-prereconcile/openshift-pipelines-scc.yaml:9]: I think that's how the scc comes out from openshift,

**Review by chmouel — COMMENTED** [review]
/lgtm

File: cmd/openshift/operator/kodata/tekton-pipeline/00-prereconcile/openshift-pipelines-scc.yaml (hunk 1 above)
Comment (nikhil-thomas) [reply cmd/openshift/operator/kodata/tekton-pipeline/00-prereconcile/openshift-pipelines-scc.yaml:9]: sure we can, 
i used what i got from
`oc get scc anyuid -o yaml | kc neat`

**Comment by nikhil-thomas** [comment]
> @nikhil-thomas this doesn't remove the old `pipeline-anyuid` though right ?

https://github.com/tektoncd/operator/pull/273
this will remove anyuid

i will cherry pick and add the cleanup to this pr

**Comment by nikhil-thomas** [comment]
> > @nikhil-thomas this doesn't remove the old `pipeline-anyuid` though right ?
> 
> #273
> this will remove anyuid
> 
> i will cherry pick and add the cleanup to this pr

I have added cleanup also to this patch. (and closed #273 )

File: cmd/openshift/operator/kodata/tekton-pipeline/00-prereconcile/openshift-pipelines-scc.yaml (hunk 1 above)
Comment (nikhil-thomas) [reply cmd/openshift/operator/kodata/tekton-pipeline/00-prereconcile/openshift-pipelines-scc.yaml]: done

**Review by vdemeester — APPROVED** [review]
/meow

**Review by pradeepitm12 — APPROVED** [review]
/lgtm

**Pull request merged**